### ISSUE TITLE
Validator does not catch invariant obs-7

### ISSUE THREAD

**Issue by wmrutten** [ewoutkramer/fhir-net-api#435]
Originally reported on Zulip by Jens Viladsen:

https://chat.fhir.org/#narrow/stream/implementers/subject/Profile.20Validation

"...the simplifier validator does not catch obs-7. If you state the same code in Observation.code as in Observation.component.code, where same means that both codes have the same coding (including coding.system and coding.code) - I guess that is what is meant by same"

Example:
https://simplifier.net/eSundhed2017Hackatho/Observation-example/~json#json

**Comment by brianpos** [comment]
Definitely something VERY weird with that example.
The fhirpath expression `component` doesn't return the list of components here.

**Comment by brianpos** [comment]
The real issue here is the evaluation of != between 2 CodeableConcepts
Should that example pass, or fail?

**Comment by ewoutkramer** [comment]
Indeed, as per the discussion (https://chat.fhir.org/#narrow/stream/implementers/subject/Profile.20Validation) - this is not a mistake in the .NET validator, but obs-7 is incorrect, and not easy to fix.

To progress, we need an update to the spec.

**Comment by ewoutkramer** [comment]
Based on this discussion, we could make a local change to obs-7 to turn it into:

    Observation.component.code.where( (coding.code = Observation.code.coding.code) and (coding.system = Observation.code.coding.system)).empty()
